### ISSUE TITLE
[feature request] Implement AV1 (draft) support

### ISSUE THREAD

**Issue by cromefire** [quietvoid/hdr10plus_tool#50]
First and foremost: Yes it is still a draft and you may of course reject this idea, but due to it already having player support (Kodi on FireTV Max), seeming relatively straight forward (is seems to be based on the already existing ITU-T T.35 Metadata/SMPTE ST 2094-40) and seeming relatively stable [for a few months](https://github.com/AOMediaCodec/av1-hdr10plus/commits/main), I feel like this would be a great addition as it allows conversion of existing h265 content to AV1 (or back if needed, although currently it doesn't seem like there's a lot of content).

Specification (draft): https://aomediacodec.github.io/av1-hdr10plus/
Sample file I have verified working (HDR10+ metadata arriving at the TV) with Kodi (Nexus) on my FireTV Max: https://github.com/AOMediaCodec/av1-hdr10plus/blob/ecdafef3eaf104135a6b28aabd960552e8209c7d/HDR10_SIZZLE_PLUS_UP-FINAL_02_AV1.mp4
AV1 bitstream parser library: https://github.com/yohhoy/av1parser

The bitstream parser above seems to already support ITU-T T.35 metadata and your code already supports the parsing of the metadata from bytes from (hopefully) the same format. So in theory it should be as easy as "only" creating some conditions to support more than one codec and some verification that it works. I have considered trying to integrate it myself, but as I lack knowledge of the format of the metadata the chances of success would be pretty low.

**Comment by quietvoid** [comment]
See https://github.com/yohhoy/av1parser/pull/14

I don't have any plan to support AV1 until the spec is finalized.

**Comment by quietvoid** [comment]
https://github.com/xiph/rav1e/pull/3000 should be working fine to generate test streams.
I'm not sure when I'll get around to this.

**Comment by cromefire** [comment]
Happy to test stuff, with Kodi, the FireTV and Panasonic OLED, I can actually test whether HDR10+ metadata is arriving at the TV (and it already works with AV1 & HDR10+).

**Comment by quietvoid** [comment]
Good to know. I can also test on a Samsung TV.

**Comment by sven-pke** [comment]
Any news about this?

**Comment by quietvoid** [comment]
Nope.

**Comment by thePublicer** [comment]
there is support to add hdr10+ to av1 see https://github.com/HandBrake/HandBrake/blob/master/NEWS.markdown

**Comment by thePublicer** [comment]
also any plans to extract hdr10plus from vp9?

**Comment by JaitinPrakash** [comment]
the official spec has reached a state where i believe you can now implement it.
https://aomediacodec.github.io/av1-hdr10plus/

**Comment by quietvoid** [comment]
The complexity isn't in the HDR10+ spec but rather parsing AV1 bitstreams.
There aren't stateful parsers that are usable so I'd probably have to implement one from scratch.

**Comment by sven-pke** [comment]
https://gstreamer.freedesktop.org/documentation/gst-plugins-bad-codecparsers/gstav1parser.html?gi-language=c I think Gstreamer is also available in Rust. Could that be an option?

**Comment by quietvoid** [comment]
GStreamer implies an additionnal dependency at runtime and I'd rather avoid that since it's more complicated for end users.

**Comment by JaitinPrakash** [comment]
handbrake seems to have a semi-functional system in place. my phone properly detects and decodes av1 hdr10+ videos, although mediainfo only reports hdr10.

**Comment by cromefire** [comment]
> my phone properly detects and decodes av1 hdr10+ videos

Do you have any good way to test that? In the past I've found you can't really tell well on the phone, except maybe with your eyes, although that's hard-ish to see.

**Comment by JaitinPrakash** [comment]
I took an hdr10+ video on my phone, encoded it with regular svt-av1 and with handbrake, and difference in color saturation, contrast, and overall vibrancy was very clear.

**Comment by JaitinPrakash** [comment]
not at the moment, and it's midnight right now, so no good bright footage. I'll get some stuff to you when the sun is up. in the mean time, HDR10+ technologies plus up video is on YouTube.

**Comment by cromefire** [comment]
Okay cool. YouTube BTW doesn't really have HDR10+ support, you can only upload HDR10+ content there and I think it will try and use that metadata to create a good approximation of the HDR10+ version.

**Comment by JaitinPrakash** [comment]
here's a basic sample file from ff pictures GmbH, an hdr mastering company. have confirmed the hdr file is hdr10+ with mediainfo, video player, and hdr10plus_tool.

https://ff.wetransfer.com/downloads/0f596c6233476b3a74a7005f2912235320210201160114/257cf4

**Comment by cromefire** [comment]
Yeah that one I know, but it's so incredibly hard to see the differences I'm always unsure if there are any or if I'm just imagining it.

**Comment by JaitinPrakash** [comment]
Handbrake SVT-AV1 HDR10+: https://files.catbox.moe/qsc8wl.mkv

FFMPEG SVT-AV1 HDR10: https://files.catbox.moe/5e3qih.webm

Samsung Camera HDR10+ HEVC Source: https://files.catbox.moe/udbnuy.mp4

**Comment by jonathanspw** [comment]
The spec is finalized and Netflix is even shipping AV1 HDR10+ content now.

**Comment by wyup** [comment]
> The spec is finalized and Netflix is even shipping AV1 HDR10+ content now.

And how does Netflix inject hdr10+.json metadata into the AV1 stream? Do they use aoenc or svt-av1-psy? AFAIK they are the only encoders that support adding the metadata at encoding?

**Comment by quietvoid** [comment]
> why isn't --hdr10plus-json option pushed to main svt-av1 so everybody can use it with ffmpeg?

do you volunteer to do it? patches welcome
either way the option requiring a file  is incompatible with FFmpeg
